lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -101,79 +90,49 @@ UsdSceneItem::Ptr UsdUndoRenameCommand::renamedItem() const
-        }
+    auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(_stage, _prim);
 
-        if (status) {
-            // The renamed scene item is a "sibling" of its original name.
-            _ufeDstItem = createSiblingSceneItem(_ufeSrcItem->path(), _usdDstPath.GetElementString());
+    if(primSpec) {
 
-            sendRenameNotification(_ufeDstItem, _ufeSrcItem->path());
-        }
+        // set prim's name
+        primSpec->SetName(_newName);
+
+         // the renamed scene item is a "sibling" of its original name.
+        _ufeDstItem = createSiblingSceneItem(_ufeSrcItem->path(), _newName);
+
+        sendRenameNotification(_ufeDstItem, _ufeSrcItem->path());
     }
     else {
-        UFE_LOG(std::string("Warning: SdfCopySpec(") +
-                _usdSrcPath.GetString() + std::string(") failed."));
+        return false;
     }
 
-    return status;
+    return true;
 }
 
 bool UsdUndoRenameCommand::renameUndo()
 {
-    // Copy the source path using CopySpec, and remove the source.
-    bool status = SdfCopySpec(_layer, _usdDstPath, _layer, _usdSrcPath);
-
-    if (status) {
-        // remove all scene description for the given path and 
-        // its subtree in the current UsdEditTarget 
-        {
-            UsdEditContext ctx(_stage, _layer);
-            status = _stage->RemovePrim(_usdDstPath);
-        }
+    auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(_stage, _ufeDstItem->prim());
+ 
+    if(primSpec) {
 
-        if (status) {
-            // create a new prim at _usdSrcPath
-            auto newPrim = _stage->DefinePrim(_usdSrcPath);
-            
-            #ifdef UFE_V2_FEATURES_AVAILABLE
-                UFE_ASSERT_MSG(newPrim, "Invalid prim cannot be inactivated.");
-            #else
-                assert(newPrim);
-            #endif
+        // set prim's name
+        primSpec->SetName(_prim.GetPath().GetElementString());
 
-            // I shouldn't have to again create a sibling sceneItem here since we already have a valid _ufeSrcItem
-            // however, I get random crashes if I don't which needs furthur investigation.  HS, 6-May-2020.
-            _ufeSrcItem = createSiblingSceneItem(_ufeDstItem->path(), _usdSrcPath.GetElementString());
+        // shouldn't have to again create a sibling sceneItem here since we already have a valid _ufeSrcItem
+        // however, I get random crashes if I don't which needs furthur investigation.  HS, 6-May-2020.
+        _ufeSrcItem = createSiblingSceneItem(_ufeDstItem->path(), _prim.GetPath().GetElementString());
 
-            sendRenameNotification(_ufeSrcItem, _ufeDstItem->path());
+        sendRenameNotification(_ufeSrcItem, _ufeDstItem->path());
 
-            _ufeDstItem = nullptr;
-        }
+        _ufeDstItem = nullptr;
     }
     else {
-        UFE_LOG(std::string("Warning: SdfCopySpec(") +
-                _usdDstPath.GetString() + std::string(") failed."));
+        return false;
     }
 
-    return status;
+    return true;
```

Comment:
On Undo:
1- Get the SdfPrimSpec to the _ufeDstItem->prim() and check it's validity.
2- Set it's name to the new name: primSpec->SetName(_prim.GetPath().GetElementString());
3- Set _ufeSrcItem and send UFE scene notification
4- Set _ufeDstItem = nullptr